src/tests/upgrades/red.ts

```diff
@@ -1,11 +1,56 @@
 import { expect } from "chai";
 
+import { GameEngine } from "../../framework/engine";
+import { createInstance } from "../../framework/mutators";
+import { InstanceState, PlayerState } from "../../framework/types";
+import { requireActivePlayer } from "../../game/helpers";
+
+import {
+  debugAction,
+  debugGotoNextTurn,
+  makeDefaultGame,
+} from "../testhelper";
+
 describe("upgrades", () => {
   describe("red", () => {
+    let $: GameEngine;
+    let P: PlayerState;
+
+    beforeEach(() => {
+      $ = makeDefaultGame();
+      P = requireActivePlayer($);
+    });
+
     describe("Bloodburn", () => {
+      let I: InstanceState;
+
+      beforeEach(() => {
+        I = createInstance($, P, $.data.lookupCard("Bloodburn"));
+      });
+
       it("can't be used on its first turn because it doesn't have " +
         "haste", () => {
-        expect(true).to.equal(false);
+        expect(I.arrivalFatigue).to.equal(true);
+
+        I.customRunes.BLOOD = 2;
```

Comment:
Is I.runes defined? Any reason why we have a "custom runes" property instead of just using "runes" for things like this?

Reply:
"custom runes" would be defined only by their name. I think runes with framework-level support should definitely be separate (things like time runes, or +/- runes)